scripts/tools/metricProgrammable.yml

```diff
@@ -0,0 +1,490 @@
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Metric Programmable parameter value information"
+version: "1.8"
+class: $tMetricProgrammable
+name: $t_metric_programmable_param_value_info_exp_t
+base: $t_base_desc_t
+members:
+    - type: $t_value_info_exp_t
+      name: "valueInfo"
+      desc: "[out] information about the parameter value"
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Metric Programmable parameter value"
+version: "1.8"
+class: $tMetricProgrammable
+name: $t_metric_programmable_param_value_exp_t
+members:
+    - type: $t_structure_type_t
+      name: stype
+      desc: "[in] type of this structure"
+    - type: "void*"
+      name: pNext
+      desc: "[in,out][optional] must be null or a pointer to an extension-specific structure (i.e. contains stype and pNext)."
+      init: nullptr
+    - type: $t_value_t
+      name: "value"
+      desc: "[in] parameter value"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Query and get the available metric programmable handles."
+version: "1.8"
+class: $tMetricProgrammable
+name: GetExp
+decl: static
+details:
+    - "Query the available programmable handles using *pCount = 0."
+    - "Returns all programmable metric handles available in the device."
+    - "The application may call this function from simultaneous threads."
+params:
+    - type: "$t_device_handle_t"
+      name: hDevice
+      desc: "[in] handle of the device"
+    - type: uint32_t*
+      name: pCount
+      desc: |
+            [in,out] pointer to the number of metric programmable handles.
+            if count is zero, then the driver shall update the value with the total number of metric programmable handles available.
+            if count is greater than the number of metric programmable handles available, then the driver shall update the value with the correct number of metric programmable handles available.
+    - type: "$t_metric_programmable_exp_handle_t*"
+      name: phMetricProgrammables
+      desc: |
+            [in,out][optional][range(0, *pCount)] array of handle of metric programmables.
+            if count is less than the number of metric programmables available, then driver shall only retrieve that number of metric programmables.
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Get the properties of the metric programmable."
+version: "1.8"
+class: $tMetricProgrammable
+name: GetPropertiesExp
+decl: static
+details:
+    - "Returns the properties of the metric programmable."
+params:
+    - type: $t_metric_programmable_exp_handle_t
+      name: hMetricProgrammable
+      desc: "[in] handle of the metric programmable"
+    - type: "$t_metric_programmable_exp_properties_t*"
+      name: pProperties
+      desc: "[out] properties of the metric programmable"
```

Comment:
multiple functions with no return error codes defined.

Reply:
I think default return error codes are added by the infra.